src/service.rs

```diff
@@ -531,20 +531,23 @@ impl Service {
                     kbucket::Entry::Present(ref mut entry, _) => {
                         if entry.value().seq() < enr_seq {
                             let enr = entry.value().clone();
-                            to_request_enr = Some(enr.into());
+                            to_request_enr = Some(enr);
                         }
                     }
                     kbucket::Entry::Pending(ref mut entry, _) => {
                         if entry.value().seq() < enr_seq {
                             let enr = entry.value().clone();
-                            to_request_enr = Some(enr.into());
+                            to_request_enr = Some(enr);
                         }
                     }
                     // don't know of the ENR, request the update
                     _ => {}
                 }
                 if let Some(enr) = to_request_enr {
-                    self.request_enr(enr, None);
+                    // TODO: check this
+                    let contact =
+                        NodeContact::try_from_enr(enr).expect("stored ENRs are contactable.");
```

Comment:
We probably should err on the side of caution and just log an error, rather than panicing here

Reply:
done